ufo2fontir/src/source.rs

```diff
@@ -1,27 +1,37 @@
 use std::{
-    collections::{BTreeMap, HashMap},
+    collections::{BTreeMap, HashMap, HashSet},
     fs,
     path::{Path, PathBuf},
 };
 
 use fontir::{
     error::{Error, WorkError},
     filestate::FileStateSet,
+    ir::DesignSpaceLocation,
     source::{Input, Paths, Source, Work},
 };
 use log::debug;
 use norad::designspace::{self, DesignSpaceDocument};
 
 pub struct DesignSpaceIrSource {
     designspace_file: PathBuf,
+    designspace_dir: PathBuf,
     ir_paths: Paths,
+    // A cache of locations, valid provided no font info file changes
+    glif_locations: (FileStateSet, HashMap<PathBuf, Vec<DesignSpaceLocation>>),
```

Comment:
why would a change in UFO fontinfo.plist invalidate the cache of glyph locations? 
oh maybe I misunderstood, seems like you used the name `font_info` (the return value of font_info_files) to mean a FileStateSet that includes not just source ufos' fontinfo.plist (which I think are irrelevant to that glif_locations cache) but also the designspace file itself (which defines those locations and thus is relevant).
Perhaps use a different name that can't be misunderstood this way

Reply:
Naming is the hardest part. Will update. 

My rationale was that font_info.plist includes information that seemed to warrant a global rebuild on change, e.g. upem. I'll add a comment as well as renaming.

Reply:
Did that help?

Reply:
yes, thanks